### ISSUE TITLE
test_invalid_keys fails on Django 1.7

### ISSUE THREAD

**Issue by MichaelAquilina** [adamchainz/django-mysql#189]
Another failure in `test_cache.py` which seems possibly related to #188 but throws a different error:

```
platform linux -- Python 3.4.3 -- py-1.4.30 -- pytest-2.7.2
django settings: settings (from environment variable)
rootdir: /home/michael/Development/django-mysql, inifile: 
plugins: django
Django version: 1.9.0.beta.1
MySQL version: 5.6.27-0ubuntu1
```

```
self = <tests.testapp.test_cache.MySQLCacheTests testMethod=test_invalid_keys>

    def test_invalid_keys(self):
        # mimic custom ``make_key`` method being defined since the default will
        # never show the below warnings
        def func(key, *args):
            return key
    
        old_func = cache.key_func
        cache.key_func = func
    
        try:
            with warnings.catch_warnings(record=True) as w:
                warnings.simplefilter("always")
                # memcached does not allow whitespace or control characters in
                # keys
                cache.set('key with spaces', 'value')
>               assert len(w) == 2
E               AssertionError: assert 3 == 2
E                +  where 3 = len([<warnings.WarningMessage object at 0x7f70532a9630>, <warnings.WarningMessage object at 0x7f70532a94a8>, <warnings.WarningMessage object at 0x7f70532c1e48>])

tests/testapp/test_cache.py:905: AssertionError
```

**Comment by adamchainz** [comment]
This sounds like the warning from #188 is appearing in this list of warnings that should be 2 warnings. Can you `--pdb` and `print` the warning messages for me?

**Comment by adamchainz** [comment]
Fixed in #198
